digital_printing/digital_printing/doctype/print_order/print_order.py

```diff
@@ -158,6 +164,84 @@ def calculate_totals(self):
 	def on_upload_complete(self):
 		self.set_missing_values()
 
+	@frappe.whitelist()
+	def create_printed_design_item(self):
+		if self.docstatus != 1:
+			frappe.throw(_("Submit the document first."))
+
+		for d in self.items:
+			if d.item_code:
+				continue
+
+			item_doc = frappe.get_doc({
+				"doctype": "Item",
+				"item_naming_by": "System Generated",
+				"item_group": "Printed Fabric",
+				"print_item_type": "Printed Design",
+				"item_name": "{0} ({1})".format(d.design_name, self.fabric_item_name),
+				"stock_uom": d.stock_uom,
+				"sales_uom": d.uom,
+				"fabric_item": self.fabric_item,
+				"process_item": self.process_item,
+				"image": d.design_image,
+				"design_name": d.design_name,
+				"design_width": d.design_width,
+				"design_height": d.design_height,
+				"design_uom": d.uom,
+				"design_gap": d.design_gap,
+				"per_wastage": d.per_wastage,
+				"design_notes": d.design_notes,
+			})
+
+			item_doc.append("uom_conversion_graph", {
+				"from_uom": "Panel",
+				"from_qty": 1,
+				"to_uom": "Meter",
+				"to_qty": d.panel_length_meter
+			})
+
+			if "Yard" in [d.stock_uom, d.uom]:
+				item_doc.append("uom_conversion_graph", {
+					"from_uom": "Yard",
+					"from_qty": 1,
+					"to_uom": "Meter",
+					"to_qty": 0.9144
+				})
+
+			item_doc.save()
+
+			frappe.db.set_value(d.doctype, d.name, "item_code", item_doc.name)
+			frappe.db.set_value(d.doctype, d.name, "item_name", item_doc.item_name)
```

Comment:
use `d.db_set()` instead of `frappe.db.set_value` .. otherwise print order controller will not be updated with these values, only the database